File: src/quick.py
#--------------------------
# Quicksort   oui oui
#--------------------------
import sys

def swap(keep, a, b):
    keep[a], keep[b]= keep[b], keep[a]

def partition(keep, low, high):
    pivot=keep[high]
    i=low
    for j in range (low, high):
        if keep[j]<= pivot:
            swap(keep, i, j)
            i += 1
    swap(keep, i, high)
    return i

def quicksort(keep, low, high):
    if low < high:
        p=partition(keep, low, high)
        quicksort(keep, low, p-1)
        quicksort(keep, p+1, high)

# Array= [0,5,9,1,2,4,8,6,3,7]
# print(Array)
# quicksort(Array, 0, 9)
# print(Array)

File: src/quick.cpp
//Quicksort but inefficient, runs into multiple of the same arrays
#include<iostream>
using namespace std;

void printarr(int keep[]){
       for(int i=0; i<10; i++)
    {
        cout << keep[i] << " ";
    }
    cout << endl;
};

void swap(int keep[], int a, int b){
    int temp=keep[a];
    keep[a]=keep[b];
    keep[b]=temp;
    printarr(keep);
};

int partition(int keep[], int low, int high){
    int pivot=keep[high];
    int i=low;
    for(int j=low; j<high; j++){
        if(keep[j]<=pivot){
            swap(keep, i, j);
            i++;
        }
    }
    swap(keep, i, high);
    return i;
};


void quicksort(int keep[], int low, int high){
    if(low < high){
        int p=partition(keep, low, high);
        quicksort(keep, low, p-1);
        quicksort(keep, p+1, high);
    }
};

int main(){
    int keep[]={6, 1, 5, 7, 2, 4, 9, 3, 8, 0};
    printarr(keep);
    quicksort(keep, 0, 9);
    printarr(keep);
    return 0;
}